test/unit/utils/nmodl_constructs.cpp

```diff
@@ -1525,42 +1525,42 @@ std::vector<DiffEqTestCase> const diff_eq_constructs{
         {
                 "GluSynapse.mod",
                 "A_AMPA' = -A_AMPA/tau_r_AMPA",
-                "A_AMPA = A_AMPA+dt*(-A_AMPA/tau_r_AMPA)",
+                "DA_AMPA = -A_AMPA/tau_r_AMPA",
                 "euler"
         },
 
         {
                 "GluSynapse.mod",
                 "m_VDCC' = (minf_VDCC-m_VDCC)/mtau_VDCC",
-                "m_VDCC = m_VDCC+dt*((minf_VDCC-m_VDCC)/mtau_VDCC)",
+                "Dm_VDCC = (minf_VDCC-m_VDCC)/mtau_VDCC",
                 "euler"
         },
 
         {
                 "GluSynapse.mod",
                 "cai_CR' = -(1e-9)*(ica_NMDA + ica_VDCC)*gamma_ca_CR/((1e-15)*volume_CR*2*FARADAY) - (cai_CR - min_ca_CR)/tau_ca_CR",
-                "cai_CR = cai_CR+dt*(-(1e-9)*(ica_NMDA + ica_VDCC)*gamma_ca_CR/((1e-15)*volume_CR*2*FARADAY) - (cai_CR - min_ca_CR)/tau_ca_CR)",
+                "Dcai_CR = -(1e-9)*(ica_NMDA + ica_VDCC)*gamma_ca_CR/((1e-15)*volume_CR*2*FARADAY) - (cai_CR - min_ca_CR)/tau_ca_CR",
                 "euler"
         },
 
         {
                 "GluSynapse.mod",
                 "effcai_GB' = -0.005*effcai_GB + (cai_CR - min_ca_CR)",
-                "effcai_GB = effcai_GB+dt*(-0.005*effcai_GB + (cai_CR - min_ca_CR))",
+                "Deffcai_GB = -0.005*effcai_GB + (cai_CR - min_ca_CR)",
                 "euler"
         },
 
         {
                 "GluSynapse.mod",
                 "Rho_GB' = ( - Rho_GB*(1-Rho_GB)*(rho_star_GB-Rho_GB) + potentiate_GB*gamma_p_GB*(1-Rho_GB) - depress_GB*gamma_d_GB*Rho_GB ) / ((1e3)*tau_GB)",
-                "Rho_GB = Rho_GB+dt*(( - Rho_GB*(1-Rho_GB)*(rho_star_GB-Rho_GB) + potentiate_GB*gamma_p_GB*(1-Rho_GB) - depress_GB*gamma_d_GB*Rho_GB ) / ((1e3)*tau_GB))",
+                "DRho_GB = ( - Rho_GB*(1-Rho_GB)*(rho_star_GB-Rho_GB) + potentiate_GB*gamma_p_GB*(1-Rho_GB) - depress_GB*gamma_d_GB*Rho_GB ) / ((1e3)*tau_GB)",
                 "euler"
         },
 
         {
                 "GluSynapse.mod",
                 "Use_GB' = (Use_d_GB + Rho_GB*(Use_p_GB-Use_d_GB) - Use_GB) / ((1e3)*tau_Use_GB)",
-                "Use_GB = Use_GB+dt*((Use_d_GB + Rho_GB*(Use_p_GB-Use_d_GB) - Use_GB) / ((1e3)*tau_Use_GB))",
+                "DUse_GB = (Use_d_GB + Rho_GB*(Use_p_GB-Use_d_GB) - Use_GB) / ((1e3)*tau_Use_GB)",
```

Comment:
@alkino : correct me if I am wrong: I don't see the `euler solution` that we create like `create_statement(fmt::format("{} = {} + dt * D{}", n, n, n));` . Possible to test that part as well? 🤔

Reply:
it is tested in codegen, because this line is in codegen